lgn-provers/src/provers/v1/query/dummy_prover.rs

```diff
@@ -27,42 +25,26 @@ impl StorageQueryProver for DummyProver
         Ok(dummy_proof(PROOF_SIZE))
     }
 
-    fn prove_full_node(
+    fn prove_row_chunks(
         &self,
-        _embedded_tree_proof: Vec<u8>,
-        _left_child_proof: Vec<u8>,
-        _right_child_proof: Vec<u8>,
+        _input: RowsChunkInput,
         _pis: &DynamicCircuitPis,
-        _is_rows_tree_node: bool,
     ) -> anyhow::Result<Vec<u8>>
     {
         Ok(dummy_proof(PROOF_SIZE))
     }
 
-    fn prove_partial_node(
+    fn prove_chunk_aggregation(
         &self,
-        _input: PartialNodeInput,
-        _embedded_proof: Vec<u8>,
-        _pis: &DynamicCircuitPis,
-    ) -> anyhow::Result<Vec<u8>>
-    {
-        Ok(dummy_proof(PROOF_SIZE))
-    }
-
-    fn prove_single_path_leaf(
-        &self,
-        _input: SinglePathLeafInput,
-        _embedded_proof: Vec<u8>,
-        _pis: &DynamicCircuitPis,
+        _chunks_proofs: &[Vec<u8>],
     ) -> anyhow::Result<Vec<u8>>
     {
         Ok(dummy_proof(PROOF_SIZE))
     }
 
-    fn prove_single_path_branch(
+    fn prove_non_existence(
```

Comment:
I dont think this one needs to exists anymore ? Non existence is just providing one input like any other no ?  @nicholas-mainardi

Reply:
For rows tree with no matching rows yes, but we still have a circuit to prove non existence of nodes in the index tree. I suppose that this variant refers to this circuit?